symengine/cwrapper.cpp

```diff
@@ -200,6 +205,34 @@ double real_double_get_d(const basic s)
     return (rcp_static_cast<const RealDouble>(s->m))->as_double();
 }
 
+#ifdef HAVE_SYMENGINE_MPFR
+
+void real_mpfr_set_d(basic s, double d, int prec)
+{
+    mpfr_t mp;
```

Comment:
that would require using this constructor: `mpfr_class(std::string s, mpfr_prec_t prec = 53, unsigned base = 10)`

For the `real_mpfr_set_str` this would be better, but for `real_mpfr_set_d` it would need to be converted into string first. What do you think?